src/Persistence/Yaml/ControlFormatter.cs

```diff
@@ -59,9 +59,9 @@ public static class ControlFormatter
         var isGroupContainer = control.Template.Name == BuiltInTemplates.GroupContainer.Name;
         var originalChildCount = control.Children.Count;
 
-        var children = control.Children
-            .Select(addZIndex)
-            .Concat(childrenToAdd)
+        var children = childrenToAdd
+            .Concat(control.Children
+            .Select(addZIndex))
```

Comment:
Style Nitpick:  
with the current indentation, the `.Select(addZIndex)` at first glance appears to apply to the result of the concatenation, not just the `control.Children` portion.

It would read clearer with something like one of the below:

```suggestion
            .Concat(control.Children
                .Select(addZIndex))
```
```suggestion
            .Concat(control.Children.Select(addZIndex))
```